shared/debug/debug.go

```diff
@@ -357,9 +357,7 @@ func startPProf(address string) {
 	http.Handle("/memsize/", http.StripPrefix("/memsize", &Memsize))
 	log.WithField("addr", fmt.Sprintf("http://%s/debug/pprof", address)).Info("Starting pprof server")
 	go func() {
-		if err := http.ListenAndServe(address, nil); err != nil {
-			log.Error("Failure in running pprof server", "err", err)
-		}
+		log.Fatal(http.ListenAndServe(address, nil))
```

Comment:
I dont think it should be fatal here, if there is a failure in the pprof server, this will crash the whole client.

Reply:
That makes sense. I reverted it back.